extras/rya.manual/src/site/markdown/rya-streams.md

```diff
@@ -0,0 +1,385 @@
+  Visibility: 
+
+  names: 
+    [name]: person  ---  [value]: urn:Alice
+    [name]: employee  ---  [value]: urn:Bob
+    [name]: business  ---  [value]: urn:TacoPlace
+  Visibility: 
+```
+
+<div id='use-cases'/>
+
+## Use Cases ##
+### Alerting ###
+An alerting system's job is to notify people, or another application, when 
+something of interest has been observed. Alert latency is the amount of time
+it takes for a system to issue the alert after the observations required to do
+so have been made. How the alerts are used determines how much latency is able 
+to be tolerated. For example, if the alerting system is used to discover when a 
+fire has started in a building, then that latency period needs to be short in 
+order to save the building. However, some systems do not require low latency.
+Status updates typically aren't time critical. An alert that you package has 
+been shipped doens't need to be timely since your package still hasn't arrived.
+
+Rya Streams is able to be used to build alerting systems. The conditions that 
+generate alerts are encapsulated within a SPARQL query, the observations that
+are watched are the RDF Statements, and the alert is a Binding Set or 
+constructed Statement that is output by the query. It being low/high latency 
+will depend on how much data needs to be processed and how performant the 
+alerting queries are. 
+
+### Inference Engine - Forward Chaining ###
+
+An inference engine that uses forward chaining starts with a set of inference 
+rules and some data. It uses the rules to generate new data until it reasons
+its way to some end goal/answer. For example, suppose that the goal is to 
+conclude the color of a pet named Fritz, given that he croaks and eats flies, 
+and that the rule base contains the following four rules:
+
+ 1. If X croaks and X eat flies - Then X is a frog
+ 2. If X chirps and X sings - Then X is a canary
+ 3. If X is a frog - Then X is green
+ 4. if X is a canary - Then X is yellow
+
+If the following two facts are evaluated using this rule set:
+* Fritz croaks
+* Fritz eats flies
+
+Then the following facts are inferred about Fritz:
+* Fritz is a frog
+* Fritz is green
+
+Rya Streams is able to be used to build an inference engine that executes that
+rule set. SPARQL queries are used to generate new Statements from the original 
+set of Statements. For example:
+
+```
+INSERT {
+    ?x <urn:isA> <urn:frog> 
+} WHERE {
+    ?x <urn:makesSound> <urn:croaks> .
+    ?x <urn:eats> <urn:flies> .
+}
+
+INSERT {
+    ?x <urn:hasColor> <urn:green>
+} WHERE {
+    ?x <urn:isA> <urn:frog>
+}
+```
+As long as the first query's results are ingested into the streams application, 
+the second query will infer that ?x is green based on the previous query's
+inferred statement. 
+
+### PCJ Maintenance ###
+
+Pre-Computed Joins are a feature within Rya where a query's results are 
+maintained and used as an index to speed up query evaluation. See the PCJ
```

Comment:
Make a link for the reference to the PCJ section: (not sure if this is right)
`[PCJ section](pcj-updater.md)`

Reply:
Done.